crates/api_common/src/utils.rs

```diff
@@ -453,14 +453,38 @@ pub fn local_site_opt_to_slur_regex(local_site: &Option<LocalSite>) -> Option<Re
 pub fn send_application_approved_email(
   user: &LocalUserView,
   settings: &Settings,
+  email: &str,
 ) -> Result<(), LemmyError> {
-  let email = &user.local_user.email.clone().expect("email");
   let lang = get_interface_language(user);
   let subject = lang.registration_approved_subject(&user.person.actor_id);
   let body = lang.registration_approved_body(&settings.hostname);
   send_email(&subject, email, &user.person.name, &body, settings)
 }
 
+pub async fn send_application_denied_email(
+  user: &LocalUserView,
+  settings: &Settings,
+  deny_reason: &Option<String>,
+  email: &str,
+) -> Result<(), LemmyError> {
+  let lang = get_interface_language(user);
+  let subject = lang.registration_denied_subject(&user.person.actor_id);
+
+  // Try to grab a reason
+  if let Some(deny_reason) = deny_reason {
+    // See if the reason is empty
+    if deny_reason.is_empty() {
+      let body = lang.registration_denied_body(&settings.hostname);
+      send_email(&subject, email, &user.person.name, &body, settings)
+    } else {
+      let body = lang.registration_denied_reason_body(deny_reason, &settings.hostname);
+      send_email(&subject, email, &user.person.name, &body, settings)
+    }
+  } else {
+    Err(LemmyError::from_message("application_email_reason_none"))
```

Comment:
I thought you still email them, even if there is no reason given? Isn't that what you're doing above?